### PR TITLE
[Serialization] Gate Clang type (de)serialization behind UseClangFunctionTypes

### PR DESCRIPTION
**Pull request by varungandhi-apple**
We shouldn't be getting source compat suite failures such as in https://ci.swift.org/job/swift-PR-source-compat-suite-debug/3182/artifact/swift-source-compat-suite/FAIL_siesta-legacy-Siesta.xcodeproj_4.0_BuildXcodeProjectTarget_Siesta-iOS_generic-platform-iOS.log at this stage because the feature is off-by-default.

### CODE REVIEW

**Comment by varungandhi-apple** [comment]
@swift-ci please clean test

File: lib/Serialization/Deserialization.cpp [hunk 1]
```diff
@@ -5799,6 +5799,9 @@ class SwiftToClangBasicReader :
 
 llvm::Expected<const clang::Type *>
 ModuleFile::getClangType(ClangTypeID TID) {
+  if (getContext().LangOpts.UseClangFunctionTypes)
+    return nullptr;
```
Comment (xymus) [inline lib/Serialization/Deserialization.cpp:5803]: Are you sure about the direction of this check? Or if we skip `getClangType` when `UseClangFunctionTypes`, what do we use instead?

**Review by xymus — APPROVED** [review]

File: lib/Serialization/Deserialization.cpp (hunk 1 above)
Comment (varungandhi-apple) [reply lib/Serialization/Deserialization.cpp:5803]: My bad, I just realized this when running tests. It's incorrect.

**Comment by varungandhi-apple** [comment]
@swift-ci please test

File: lib/Serialization/Deserialization.cpp (hunk 1 above)
Comment (varungandhi-apple) [reply lib/Serialization/Deserialization.cpp:5803]: Fixed polarity of check.

**Comment by varungandhi-apple** [comment]
@swift-ci please test

**Comment by varungandhi-apple** [comment]
@swift-ci please clean test macOS

**Review by rjmccall — APPROVED** [review]
LGTM

**Pull request merged**